src/main.cpp

```diff
@@ -720,26 +739,49 @@ int main(int argc, char**argv)
   BillingHandlerConfig* cfg = new BillingHandlerConfig();
   PeerMessageSenderFactory* factory = new PeerMessageSenderFactory(options.billing_realm);
 
+  // Create a connection to Chronos.
   std::string port_str = std::to_string(options.http_port);
 
-  // We want Chronos to call back to its local sprout instance so that we can
-  // handle Ralfs failing without missing timers.
-  int http_af = AF_INET;
-  std::string chronos_callback_addr = "127.0.0.1:" + port_str;
-  std::string local_chronos = "127.0.0.1:7253";
-  if (is_ipv6(options.http_address))
+  std::string chronos_service;
+  std::string chronos_callback_addr;
+  int http_af;
+
+  if (options.chronos_hostname == "" || options.ralf_hostname == "")
   {
-    http_af = AF_INET6;
-    chronos_callback_addr = "[::1]:" + port_str;
-    local_chronos = "[::1]:7253";
+    // We want Chronos to call back to its local Ralf instance so that we can
+    // handle Ralfs failing without missing timers.
+    chronos_service = "127.0.0.1:7253";
+    http_af = AF_INET;
+
+    if (is_ipv6(options.http_address))
+    {
+      chronos_callback_addr = "[::1]:" + port_str;
+    }
+    else
+    {
+      chronos_callback_addr = "127.0.0.1:" + port_str;
+    }
+  }
+  else
+  {
+    chronos_service = options.chronos_hostname + ":7253";
+
+    http_af = is_ipv6(options.chronos_hostname) ? AF_INET6 : AF_INET;
+
+    chronos_callback_addr = options.ralf_hostname + ":" + port_str;
```

Comment:
`ralf_hostname` contains a port so doesn't this end up as `ralf.example.com:10888:10888`?